Source/EasyNetQ/IMessageSerializationStrategy.cs

```diff
@@ -18,23 +21,24 @@ public interface IMessageSerializationStrategy
         /// <param name="properties">The properties</param>
         /// <param name="body">The body</param>
         /// <returns></returns>
-        IMessage DeserializeMessage(MessageProperties properties, byte[] body);
+        IMessage DeserializeMessage(MessageProperties properties, ReadOnlyMemory<byte> body);
     }
 
     /// <summary>
     ///     Represents a serialized message
     /// </summary>
-    public readonly struct SerializedMessage
+    public readonly struct SerializedMessage: IDisposable
```

Comment:
is this the part that will help disposing the buffer coming from method like BasicGet?

Reply:
It will help on publisher side to dispose memory allocated for serialization purposes(message -> byte[]).

So we 1) serialise message(buffer is allocated from the pool) and `SerializedMessage` is created 2) write it to `IModel` via `BasicPublish` 3) dispose  `SerializedMessage` to return buffer back to the pool